src/Redux/SuperAdmins/actions.js

```diff
@@ -0,0 +1,116 @@
+import {
+  GET_SUPERADMINS_PENDING,
+  GET_SUPERADMINS_SUCCESS,
+  GET_SUPERADMINS_ERROR,
+  DELETE_SUPERADMINS_PENDING,
+  DELETE_SUPERADMINS_SUCCESS,
+  DELETE_SUPERADMINS_ERROR,
+  GET_SUPERADMINS_BY_ID_PENDING,
+  GET_SUPERADMINS_BY_ID_SUCCESS,
+  GET_SUPERADMINS_BY_ID_ERROR,
+  CREATE_SUPERADMINS_PENDING,
+  CREATE_SUPERADMINS_SUCCESS,
+  CREATE_SUPERADMINS_ERROR,
+  EDIT_SUPERADMIN_PENDING,
+  EDIT_SUPERADMIN_SUCCESS
+} from './constants';
+
+export const getSuperAdminsPending = () => {
+  return {
+    type: GET_SUPERADMINS_PENDING
+  };
+};
+
+export const getSuperAdminsSuccess = (data) => {
+  return {
+    type: GET_SUPERADMINS_SUCCESS,
+    payload: data
+  };
+};
+
+export const getSuperAdminsError = (error) => {
+  return {
+    type: GET_SUPERADMINS_ERROR,
+    payload: error
+  };
+};
+
+export const getSuperAdminsIdPending = () => {
+  return {
+    type: GET_SUPERADMINS_BY_ID_PENDING
+  };
+};
+
+export const getSuperAdminsIdSuccess = (id) => {
+  return {
+    type: GET_SUPERADMINS_BY_ID_SUCCESS,
+    payload: id
+  };
+};
+
+export const getSuperAdminsIdError = (error) => {
+  return {
+    type: GET_SUPERADMINS_BY_ID_ERROR,
+    payload: error
+  };
+};
+
+export const createSuperAdminPending = () => {
+  return {
+    type: CREATE_SUPERADMINS_PENDING
+  };
+};
+
+export const createSuperAdminSuccess = (data) => {
+  return {
+    type: CREATE_SUPERADMINS_SUCCESS,
+    payload: data
+  };
+};
+
+export const createSuperAdminError = (error) => {
+  return {
+    type: CREATE_SUPERADMINS_ERROR,
+    payload: error
+  };
+};
+
+export const editSuperAdminPending = () => {
+  return {
+    type: EDIT_SUPERADMIN_PENDING
+  };
+};
+
+export const editSuperAdminSuccess = (updatedSuperAdmin) => {
```

Comment:
here you need the ID to validate, you need to validate with redux dev tools extension, here you can watch what is happening every moment
```suggestion
export const editSuperAdminSuccess = (updatedSuperAdmin, id) => {
```

Reply:
i dont need the id in the accion, becouse i dont use the id for execute nothing with the reducer.

Reply:
However i use the id in the thunk for execute the request in the correct way.

Reply:
if it pass the redux dev tools it's fine! I did that comment because if I didn't pass the ID it didn't work fine when I test in dev tools